kusto/kql/query_parameters.go

```diff
@@ -0,0 +1,67 @@
+package kql
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+type ParamVals struct {
+	paramType string
+	value     string
+}
+
+type StatementQueryParameters struct {
+	parameters map[string]ParamVals
+}
+
+var buildPool = sync.Pool{
+	New: func() interface{} {
+		return new(strings.Builder)
+	},
+}
+
+func NewStatementQueryParameters() *StatementQueryParameters {
+	return &StatementQueryParameters{parameters: make(map[string]ParamVals)}
+}
+func (q *StatementQueryParameters) addBase(key string, paramType string, value fmt.Stringer) *StatementQueryParameters {
+	q.parameters[key] = ParamVals{paramType, value.String()}
+	return q
+}
+
+// note - due to the psuedo-random nature of maps, the declaration string might be ordered differently for different runs.
+// might crash the test in those times.
+func (q *StatementQueryParameters) ToDeclarationString() string {
+	const (
+		declare   = "declare query_parameters("
+		closeStmt = ");"
+	)
+
+	if len(q.parameters) == 0 {
+		return ""
+	}
+
+	build := buildPool.Get().(*strings.Builder)
+	build.Reset()
+	defer buildPool.Put(build)
+
+	build.WriteString(declare)
+
+	for key, paramVals := range q.parameters {
```

Comment:
x = false 
for 1 .. n 
    if x
       add ","
    x = true

Reply:
I assume then it's putting the comma before the string?
@ronmonetaMicro you didn't implement this

Reply:
i've implemented it via lengths, instead of boolean flags + lengths

Reply:
@yogilad WDYT?

Reply:
Slightly more complex, but all in all ok